File: checkio_js/elementary/javascript/splitPairs.js
import assert from "assert";

function splitPairs(text) {
    let arr = []
    for (let i=0; i<text.length; i+=2){
      if (text.substr(i, 2).length < 2) {
        arr.push(text.substr(i, 2) + '_')
      } else {
        arr.push(text.substr(i, 2))
      }
    }
    return arr;
}

console.log('Example:');
console.log(splitPairs('abcd'));

// These "asserts" are used for self-checking
assert.deepEqual(splitPairs('abcd'), ['ab', 'cd']);
assert.deepEqual(splitPairs('abc'), ['ab', 'c_']);
assert.deepEqual(splitPairs('abcdf'), ['ab', 'cd', 'f_']);
assert.deepEqual(splitPairs('a'), ['a_']);
assert.deepEqual(splitPairs(''), []);

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/elementary/python/splitPairs.py


def splitPairs(text):
    arr = []
    for i in range(0, len(text), 2):
        if len(text[i:i+2]) == 1:
            arr.append(text[i:i+2] + '_')
        else:
            arr.append(text[i:i+2])
    return arr

if __name__ == '__main__':
    print('Example:')
    print(splitPairs('abcd'))

# These "asserts" are used for self-checking
assert splitPairs('abcd') == ['ab', 'cd']
assert splitPairs('abc') == ['ab', 'c_']
assert splitPairs('abcdf') == ['ab', 'cd', 'f_']
assert splitPairs('a') == ['a_']
assert splitPairs('') == []

print ("Coding complete? Click 'Check' to earn cool rewards!")
